File: python/3.py
def longest_common_prefix(strs):
   if not strs:
       return ""

   first_str = strs[0]
   for i in range(1, len(strs)):
       while not strs[i].startswith(first_str):
           first_str = first_str[:-1]

   return first_str

strs = ["flower","flow","flight"]
print(f"\n\n\n\nCorrect {longest_common_prefix(strs)}\n\n")

File: c++/3.cpp
#include <iostream>
#include <vector>
#include <string>

std::string longestCommonPrefix(std::vector<std::string> strs) {
   if (strs.empty())
       return "";

   std::string prefix = strs[0];
   for (int i = 1; i < strs.size(); i++) {
       while (strs[i].find(prefix) != 0) {
           prefix = prefix.substr(0, prefix.length() - 1);
       }
   }

   return prefix;
}

int main() {
   std::vector<std::string> strs = {"flower", "flow", "flight"};
   std::cout << "\n\n\n\nCorrect " << longestCommonPrefix(strs) << "\n\n";
   return 0;
}
